opm/grid/MinpvProcessor.hpp

```diff
@@ -332,6 +353,28 @@ namespace Opm
                             kk = kk_iter;
                         }
                     }
+                    else
+                    {
+                        if (kk < dims_[2] - 1 && (actnum.empty() || actnum[c]) && pv[c] > minpvv[c] &&
+                            multz(c) != 0.0)
+                        {
+                            // Check whether there is a gap to the neighbor below whose thickness is less
+                            // than MAX_GAP. In that case we need to create an NNC.
+                            int kk_below = kk + 1;
+                            int c_below = ii + dims_[0] * (jj + dims_[1] * kk_below);
+
+                            if ((actnum.empty() || actnum[c_below]) && pv[c_below] > minpvv[c_below])
+                            {
+                                // Check MAX_GAP threshold
+                                std::array<double, 8> cz = getCellZcorn(ii, jj, kk, zcorn);
+                                std::array<double, 8> cz_below = getCellZcorn(ii, jj, kk_below, zcorn);
+
+                                if (computeGap(cz, cz_below) < max_gap) {
+                                    result.add_nnc(c, c_below);
+                                }
+                            }
```

Comment:
Yeah, this is quite some overkill as we will add an NNC where there often will be a vertical connection between two active cells anyway.
This should probably move into the direct CpGrid generation where we decide whether there is a face between two cells based on whether their vertical faces coincide.

Reply:
Alternatively, we could check here whether there is actually a gap to close with an NNC here. That would keep PINCH separated from the topology construction.

Reply:
Fixed in [be8b720](https://github.com/OPM/opm-grid/pull/658/commits/be8b720f6fe0d59e2cc829b7a749d46fafe273a0) where we check whether there is actually a gap.